src/app/child-dev-project/children/notes-list/notes-list.component.ts

```diff
@@ -30,36 +33,35 @@ export class NotesComponent implements OnInit {
       (v) => '', 'md'),
   ];
 
-
   constructor(private route: ActivatedRoute,
               private childrenService: ChildrenService,
               private sessionService: SessionService,
-              private datePipe: DatePipe) {
-  }
+              private datePipe: DatePipe,
+              private entityMapperService: EntityMapperService) {}
 
   ngOnInit() {
     this.route.paramMap.subscribe(params => {
       this.childId = params.get('id');
+    });
 
-      this.childrenService.getNotesOfChild(this.childId)
-        .subscribe(results => this.records = results.sort((a, b) => {
-          return (b.date ? b.date.valueOf() : 0) - (a.date ? a.date.valueOf() : 0); } ));
+    this.entityMapperService.loadType<NoteModel>(NoteModel).then(notes => {
```

Comment:
Loading every note and doing in-memory filtering may become a performance issue. `ChildrenService.getNotesOfChild()` uses a pouchdb database index and should (in theory) be the better choice.

Did you have any particular issues that pushed you to change this here?

Reply:
I had an issue with this but could resolve this in another way. Until now I thought that these two ways of getting the data were equal in performance. If that's not the case, I will revert these changes as well

Reply:
Looking at the code right now, the issue was not resolved. `ChildrenService` filters if the children-array of the Note contains the child-id (which is never true since the Array is of type `AttendanceModel`). Is it possible to change the database quarry to check if any of it's `AttendanceModel`s contains the id?

Reply:
For the PouchDb/CouchDB based Database queries should be more efficient, the database then only loads the required documents and returns them (and hopefully the developers of pouchdb have figured out a way to do that reasonably efficient).

The query index is also created in the ChildrenService:
https://github.com/Aam-Digital/ndb-core/blob/134d808a86fde5983c81e93f2ac1e6d0a6bba98b/src/app/child-dev-project/children/children.service.ts#L229-L243

Beware, unfortunately the function has to be a string containing the javascript function (to prevent the build optimization from messing this up). You can also refer to the PouchDB guide about how those query functions work: https://pouchdb.com/guides/queries.html

Basically you'll just have to change that function to `emit()` the childId from its new nested place.